WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java

```diff
@@ -328,8 +329,18 @@ private void loadAvatar(final String avatarUrl, String injectFilePath) {
                     newAvatarUploaded ? injectFilePath : avatarUrl, new RequestListener<Drawable>() {
                         @Override
                         public void onLoadFailed(@Nullable Exception e) {
-                            ToastUtils.showToast(getActivity(), R.string.error_refreshing_gravatar,
-                                    ToastUtils.Duration.SHORT);
+                            final String appLogMessage = "onLoadFailed while loading Gravatar image!";
+                            if (e == null) {
+                                AppLog.e(T.MAIN, appLogMessage + " e == null");
+                            } else {
+                                AppLog.e(T.MAIN, appLogMessage, e);
+                            }
+
+                            // For some reason, the Activity can be null so, guard for it. See #8590.
+                            if (getActivity() != null) {
```

Comment:
Maybe the activity can be null if the user already logged out? In that case `WPMainActivity` is gone when the network response returns to the fragment.

Wondering why we're not checking if the fragment is attached or not here, and `onResourceReady`.

Reply:
> Maybe the activity can be null if the user already logged out?

Could be, yeah. Do you perhaps have something in mind to handle that case differently?

> Wondering why we're not checking if the fragment is attached or not here, and `onResourceReady`.

Oh, you think it could make a difference for `onLoadFailed()`? I guess we'd still want the exception to be logged and then guard the Toast with the `isAdded()`. I can go ahead and do the chance if you prefer that.

Regarding `onResourceReady()`, it doesn't seem to need access to the Activity so, not sure it needs to be guarded. Do you see a need there @daniloercoli?

Reply:
> Could be, yeah. Do you perhaps have something in mind to handle that case differently?

Checking for `null` activity is ok, since we're already removing pending network requests on logout. Maybe this particular request was already running on logout and the clear method didn't stop it. Anyway, your code is good to go.

Re-checked the code, and it seems we don't need to add those `isAdded` guards I mentioned in the comment above.